### ISSUE TITLE
Change TTL value of a TCP segment/TCP connection

### ISSUE THREAD

**Issue by ManiAm** [inet-framework/inet#393]
I am trying to set the hop limit (TTL) for an IP packet that is carrying a BGP message.

```
    Packet *pk = new Packet("BgpOpen");
    pk->addTagIfAbsent<HopLimitReq>()->setHopLimit(1);
    pk->insertAtFront(openMsg);
    _info.socket->send(pk);  // send to TCP
```
I am adding the HopLimitReq tag to the BGP message and then send the packet to TCP. However, TCP does not seem to let the IP protocol know about this tag, because IP is stil using the default TTL value for the IP packet. So I was wondering if this is a bug, or I am doing it wrong. Can I set the TTL value for each individual TCP segments? or can I specify the TTL value for the whole TCP connection?

**Comment by levy** [comment]
Since TCP is a connection oriented protocol, you can't really expect packet tags to have the same effect as with UDP, for example. Application packets are split and merged depending on TCP flow control, so I guess a TTL value attached to a packet is not so useful. Unfortunately, you can't easily set tags on outgoing TCP segments, becase there's no customization point for that in TCP. Nevertheless, TCP connections could support attaching packet tags, which would in turn go down to IP (e.g. TTL). It's just not implemented yet.

**Comment by ManiAm** [comment]
>>> Nevertheless, TCP connections could support attaching packet tags, which would in turn go down to IP (e.g. TTL). It's just not implemented yet.

It makes sense.

**Comment by ManiAm** [comment]
implemented in #462 :)
